pkg/clusterapi/aws/actuator.go

```diff
@@ -0,0 +1,403 @@
+
+	// Annotation used to store serialized ClusterVersion resource in MachineSet
+	clusterVersionAnnotation = "cluster-operator.openshift.io/cluster-version"
+
+	// Instance ID annotation
+	instanceIDAnnotation = "cluster-operator.openshift.io/aws-instance-id"
+)
+
+// Actuator is the driver struct for holding AWS machine information
+type AWSActuator struct {
+	kubeClient    *kubernetes.Clientset
+	clusterClient *clusterclient.Clientset
+	codecFactory  serializer.CodecFactory
+}
+
+// NewAWSDriver returns an empty AWSDriver object
+func NewActuator(kubeClient *kubernetes.Clientset, clusterClient *clusterclient.Clientset) *AWSActuator {
+	actuator := &AWSActuator{
+		kubeClient:    kubeClient,
+		clusterClient: clusterClient,
+		codecFactory:  coapi.Codecs,
+	}
+	return actuator
+}
+
+// Actuator controls machines on a specific infrastructure. All
+// methods should be idempotent unless otherwise specified.
+type Actuator interface {
+	// Create the machine.
+	Create(*clusterv1.Cluster, *clusterv1.Machine) error
+	// Delete the machine.
+	Delete(*clusterv1.Machine) error
+	// Update the machine to the provided definition.
+	Update(c *clusterv1.Cluster, machine *clusterv1.Machine) error
+	// Checks if the machine currently exists.
+	Exists(*clusterv1.Machine) (bool, error)
+}
+
+// Create runs a new EC2 instance
+func (a *AWSActuator) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	result, err := a.CreateMachine(cluster, machine)
+	if err != nil {
+		return err
+	}
+	machineCopy := machine.DeepCopy()
+	machineCopy.Annotations[instanceIDAnnotation] = *(result.Instances[0].InstanceId)
+	_, err = a.clusterClient.ClusterV1alpha1().Machines(machineCopy.Namespace).Update(machineCopy)
+	return err
+}
+
+func (a *AWSActuator) CreateMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (*ec2.Reservation, error) {
+	// Extract cluster operator cluster
+	coCluster, err := a.clusterOperatorCluster(cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	if coCluster.Spec.Hardware.AWS == nil {
+		return nil, fmt.Errorf("Cluster does not contain an AWS hardware spec")
+	}
+
+	region := coCluster.Spec.Hardware.AWS.Region
+	svc, err := a.createSVC(region)
+	if err != nil {
+		return nil, err
+	}
+
+	// For now, we store the machineSet resource in the machine.
+	// It really should be the machine resource, but it's not yet
+	// fleshed out in the Cluster Operator API
+	coMachineSet, coClusterVersion, err := a.clusterOperatorMachineSet(machine)
+	if err != nil {
+		return nil, err
+	}
+
+	if coClusterVersion.Spec.VMImages.AWSImages == nil {
+		return nil, fmt.Errorf("Cluster version does not contain AWS images")
+	}
+
+	// Get AMI to use
+	amiName := amiForRegion(coClusterVersion.Spec.VMImages.AWSImages.RegionAMIs, region)
+	if len(amiName) == 0 {
+		return nil, fmt.Errorf("Cannot determine AMI name from cluster version %#v and region %s", coClusterVersion, region)
+	}
+	imageIds := []*string{aws.String(amiName)}
```

Comment:
In general we should get some good logging in here, doesn't have to be right away though as I know this is very preliminary.